collector/transport_zone_collector.go

```diff
@@ -0,0 +1,136 @@
+package collector
+
+import (
+	"nsxt_exporter/client"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	nsxt "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/manager"
+)
+
+func init() {
+	registerCollector("transport_zone", newTransportZoneCollector)
+}
+
+type transportZoneCollector struct {
+	transportZoneClient client.TransportZoneClient
+	logger              log.Logger
+
+	transportZoneLogicalPort   *prometheus.Desc
+	transportZoneLogicalSwitch *prometheus.Desc
+	transportZoneTransportNode *prometheus.Desc
+	transportZoneDegradedNode  *prometheus.Desc
+	transportZoneDownNode      *prometheus.Desc
+	transportZoneUnknownNode   *prometheus.Desc
+	transportZoneUpNode        *prometheus.Desc
+}
+
+func newTransportZoneCollector(apiClient *nsxt.APIClient, logger log.Logger) prometheus.Collector {
+	nsxtClient := client.NewNSXTClient(apiClient, logger)
+	transportZoneLogicalPort := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "transport_zone", "total_logical_port"),
+		"Total number of logical port in transport zone",
+		[]string{"id", "name"},
+		nil,
+	)
+	transportZoneLogicalSwitch := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "transport_zone", "total_logical_switch"),
```

Comment:
total has to be suffix according to naming guideline
https://prometheus.io/docs/practices/naming/#metric-names